src/metpy/calc/thermo.py

```diff
@@ -1632,6 +1632,45 @@ def saturation_equivalent_potential_temperature(pressure, temperature):
     return units.Quantity(th_es, units.kelvin)
 
 
+@exporter.export
+@preprocess_and_wrap(
+    wrap_like='temperature',
+    broadcast=('pressure', 'temperature', 'dewpoint')
+)
+@check_units('[pressure]', '[temperature]', '[temperature]')
+def wetbulb_potential_temperature(pressure, temperature, dewpoint):
+    r"""Calculate wet-bulb potential temperature.
+
+    This calculation must be given an air parcel's pressure, temperature, and dewpoint.
+    The implementation uses the formula outlined in [DaviesJones2008]_:
+
+    First, theta-e is calculated
+    then use the formula from [DaviesJones2008]_
+
+    Parameters
+    ----------
+    pressure: `pint.Quantity`
+        Total atmospheric pressure
+
+    temperature: `pint.Quantity`
+        Temperature of parcel
+
+    dewpoint: `pint.Quantity`
+        Dewpoint of parcel
+
+    Returns
+    -------
+    `pint.Quantity`
+        wet-bulb potential temperature of the parcel
+
+    """
+    theta_e = equivalent_potential_temperature(pressure, temperature, dewpoint).magnitude
+    x = theta_e / 273.15
```

Comment:
We can be a little more tight with units here:
```suggestion
    theta_e = equivalent_potential_temperature(pressure, temperature, dewpoint)
    x = theta_e.m_as('kelvin') / 273.15
```